modules/imgproc/src/thresh.cpp

```diff
@@ -125,243 +123,241 @@ thresh_8u( const Mat& _src, Mat& _dst, uchar thresh, uchar maxval, int type )
     }
 #endif
 
-    switch( type )
-    {
-    case THRESH_BINARY:
-        for( i = 0; i <= thresh; i++ )
-            tab[i] = 0;
-        for( ; i < 256; i++ )
-            tab[i] = maxval;
-        break;
-    case THRESH_BINARY_INV:
-        for( i = 0; i <= thresh; i++ )
-            tab[i] = maxval;
-        for( ; i < 256; i++ )
-            tab[i] = 0;
-        break;
-    case THRESH_TRUNC:
-        for( i = 0; i <= thresh; i++ )
-            tab[i] = (uchar)i;
-        for( ; i < 256; i++ )
-            tab[i] = thresh;
-        break;
-    case THRESH_TOZERO:
-        for( i = 0; i <= thresh; i++ )
-            tab[i] = 0;
-        for( ; i < 256; i++ )
-            tab[i] = (uchar)i;
-        break;
-    case THRESH_TOZERO_INV:
-        for( i = 0; i <= thresh; i++ )
-            tab[i] = (uchar)i;
-        for( ; i < 256; i++ )
-            tab[i] = 0;
-        break;
-    default:
-        CV_Error( CV_StsBadArg, "Unknown threshold type" );
-    }
-
+    int j_scalar = 0;
+    const uchar* src = _src.ptr();
+    uchar* dst = _dst.ptr();
 #if CV_SSE2
-    if( checkHardwareSupport(CV_CPU_SSE2) )
+    if( (roi.width >= 8) && checkHardwareSupport(CV_CPU_SSE2) )
     {
         __m128i _x80 = _mm_set1_epi8('\x80');
         __m128i thresh_u = _mm_set1_epi8(thresh);
         __m128i thresh_s = _mm_set1_epi8(thresh ^ 0x80);
         __m128i maxval_ = _mm_set1_epi8(maxval);
-        j_scalar = roi.width & -8;
 
-        for( i = 0; i < roi.height; i++ )
+        switch( type )
         {
-            const uchar* src = _src.ptr() + src_step*i;
-            uchar* dst = _dst.ptr() + dst_step*i;
-
-            switch( type )
+        case THRESH_BINARY:
+            for( int i = 0; i < roi.height; i++, src += src_step, dst += dst_step )
             {
-            case THRESH_BINARY:
-                for( j = 0; j <= roi.width - 32; j += 32 )
+                for( j_scalar = 0; j_scalar <= roi.width - 32; j_scalar += 32 )
                 {
                     __m128i v0, v1;
-                    v0 = _mm_loadu_si128( (const __m128i*)(src + j) );
-                    v1 = _mm_loadu_si128( (const __m128i*)(src + j + 16) );
+                    v0 = _mm_loadu_si128( (const __m128i*)(src + j_scalar) );
```

Comment:
is there a reason to rename j to j_scalar? such a rename does not seem absolutely necessary and it creates really a lot of noise in the patch; could you please re-do your patch without this change?